common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -46,16 +46,22 @@
   private static final boolean unaligned;
   static {
     boolean _unaligned;
+    String arch = System.getProperty("os.arch", "");
     // use reflection to access unaligned field
     try {
       Class<?> bitsClass =
         Class.forName("java.nio.Bits", false, ClassLoader.getSystemClassLoader());
       Method unalignedMethod = bitsClass.getDeclaredMethod("unaligned");
       unalignedMethod.setAccessible(true);
-      _unaligned = Boolean.TRUE.equals(unalignedMethod.invoke(null));
+
+      //Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231)
+      if(arch.matches("^(ppc64le|ppc64)$")){
```

Comment:
I think you can perform this check up top? you don't need to do all the reflection here if the arch matches. Also is it true that anything starting with "ppc64" should be special-cased?